sessionctx/variable/session.go

```diff
@@ -643,7 +645,19 @@ func (s *SessionVars) SetSystemVar(name string, val string) error {
 	case TiDBCurrentTS, TiDBConfig:
 		return ErrReadOnly
 	case TiDBMaxChunkSize:
-		s.MaxChunkSize = tidbOptPositiveInt32(val, DefMaxChunkSize)
+		maxChunkSize := tidbOptPositiveInt32(val, DefMaxChunkSize)
+		if s.InitChunkSize > maxChunkSize {
+			s.MaxChunkSize = mathutil.Max(s.InitChunkSize, DefMaxChunkSize)
+			s.StmtCtx.AppendWarning(errors.Errorf("Try to set TiDB MaxChunkSize to %d small than InitChunkSize %d so force use %d",
+				maxChunkSize, s.InitChunkSize, s.MaxChunkSize))
+		} else {
+			s.MaxChunkSize = maxChunkSize
+		}
+	case TiDBInitChunkSize:
+		s.InitChunkSize = tidbOptPositiveInt32(val, DefInitChunkSize)
+		if s.InitChunkSize > s.MaxChunkSize {
```

Comment:
return error for `else`?

Reply:
InitChunkSize small than MaxChunkSize is normal situation